pkg/kubelet/images/image_gc_manager.go

```diff
@@ -113,22 +113,25 @@ type imageCache struct {
 	images []container.Image
 }
 
-// set updates image cache.
+// set sorts the input list and updates image cache.
 func (i *imageCache) set(images []container.Image) {
 	i.Lock()
 	defer i.Unlock()
+	// make a copy to avoid potential data race
```

Comment:
I think we can't avoid the data race, because the image list is passed in, and the caller may concurrently do some operation on it.

However, I do agree that copying it here is better. :)

Reply:
Ah you're right. I updated the comment to document that the caller should not reference the list again (similar to the comments for [client-go cache](https://github.com/kubernetes/kubernetes/blob/f8ff8f44206ff4dd9b58386d96462b01a3d79fb1/staging/src/k8s.io/client-go/tools/cache/store.go#L231-L233) and [fifo queue](https://github.com/kubernetes/kubernetes/blob/f8ff8f44206ff4dd9b58386d96462b01a3d79fb1/staging/src/k8s.io/client-go/tools/cache/fifo.go#L313-L315))
```
 // set sorts the input list and updates image cache.
+// 'i' takes ownership of the list, you should not reference the list again
+// after calling this function.
 func (i *imageCache) set(images []container.Image) {
        i.Lock()
        defer i.Unlock()
-       // make a copy to avoid potential data race
+       // Make a copy of the input list. This doesn't completely avoid
+       // potential data race. The caller should not reference the list again
+       // after calling this function.
```